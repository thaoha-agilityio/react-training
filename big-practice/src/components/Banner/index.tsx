import { memo } from 'react';
import { Link } from 'react-router-dom';
import { Box, Breadcrumb, BreadcrumbItem, Text, BreadcrumbLink, VStack } from '@chakra-ui/react';
import { ChevronRightIcon } from '@chakra-ui/icons';

// Images
import banner from '@assets/photos/banner.png';
import bannerWepb from '@assets/photos/banner.webp';

// Types
import { IMenuItem } from '@types';

// Constants
import { MENU } from '@constants';

type BannerProps = {
  title: string;
  breadcrumbItems: IMenuItem;
};

export const Banner = memo(({ title, breadcrumbItems }: BannerProps) => {
  return (
    <Box
      bgImage={[`url(${banner})`, `url(${bannerWepb})`]}
      bgSize='cover'
      w='100%'
      height='316px'
      bgPos='center'
      textAlign='center'
    >
      <VStack pt='100px'>
        <Text fontSize='4xl' fontWeight='medium' data-testid='title'>
          {title}
        </Text>
        <Breadcrumb spacing='8px' separator={<ChevronRightIcon color='gray.500' />}>
          <BreadcrumbItem>
            <BreadcrumbLink as={Link} to={MENU[0].path}>
              {MENU[0].title}
            </BreadcrumbLink>
          </BreadcrumbItem>
          <BreadcrumbItem isCurrentPage data-testid='breadcrumb-item'>
            <BreadcrumbLink as={Link} to={breadcrumbItems.path}>
              {breadcrumbItems.title}
            </BreadcrumbLink>
          </BreadcrumbItem>
        </Breadcrumb>
      </VStack>
    </Box>
  );
});
