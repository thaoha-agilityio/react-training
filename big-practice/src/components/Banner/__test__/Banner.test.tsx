import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

//Constants
import { DATA_CRUMBS } from '@constants';

// Component
import Banner from '../index';

const bannerProps = {
  title: 'Shop',
  crumbs: DATA_CRUMBS,
};

describe('Banner Component', () => {
  it('displays the title correctly', () => {
    const { getByTestId } = render(
      <Router>
        <Banner {...bannerProps} />
      </Router>,
    );

    const titleElement = getByTestId('title');
    expect(titleElement).toHaveTextContent(bannerProps.title);
  });

  it('displays the breadcrumb correctly', () => {
    const { getByTestId } = render(
      <Router>
        <Banner {...bannerProps} />
      </Router>,
    );
    const titleElement = getByTestId('breadcrumb-item');
    expect(titleElement.textContent).toEqual('Edit Product');
  });
});
