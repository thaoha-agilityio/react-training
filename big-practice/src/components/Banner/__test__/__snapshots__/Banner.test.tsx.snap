// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component displays the title correctly 1`] = `
<div>
  <div
    class="css-i9nokg"
  >
    <div
      class="chakra-stack css-103yzil"
    >
      <p
        class="chakra-text css-18xzx14"
        data-testid="title"
      >
        Shop
      </p>
      <nav
        aria-label="breadcrumb"
        class="chakra-breadcrumb css-0"
      >
        <ol
          class="chakra-breadcrumb__list css-70qvj9"
        >
          <li
            class="chakra-breadcrumb__list-item css-18biwo"
          >
            <a
              class="chakra-breadcrumb__link css-0"
              href="/shop"
            >
              Shop
            </a>
            <span
              class="css-syhbcg"
              role="presentation"
            >
              <svg
                class="chakra-icon css-rozz42"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </li>
          <li
            class="chakra-breadcrumb__list-item css-18biwo"
            data-testid="breadcrumb-item"
          >
            <span
              aria-current="page"
              class="chakra-breadcrumb__link css-0"
            >
              Edit Product
            </span>
          </li>
        </ol>
      </nav>
    </div>
  </div>
</div>
`;
